cpp/include/raft/linalg/rsvd.cuh

```diff
@@ -137,6 +139,131 @@ void rsvdPerc(const raft::handle_t& handle,
                    stream);
 }
 
+/**
+ * @defgroup rsvd Randomized Singular Value Decomposition
+ * @{
+ */
+
+/**
+ * @brief randomized singular value decomposition (RSVD) on the column major
+ * input matrix (Jacobi-based), by specifying no. of PCs and
+ * upsamples directly
+ * @param[in] handle raft::handle_t
+ * @param[in] M input raft::device_matrix_view with layout raft::col_major of shape (M, N)
+ * @param[out] S_vec singular values raft::device_vector_view of shape (K)
+ * @param[in] p no. of upsamples
+ * @param[in] use_bbt whether use eigen decomposition in computation or not
+ * @param[in] use_jacobi whether to jacobi solver for decomposition
+ * @param[in] tol tolerance for Jacobi-based solvers
+ * @param[in] max_sweeps maximum number of sweeps for Jacobi-based solvers
+ * @param[out] U optional left singular values of raft::device_matrix_view with layout
+ * raft::col_major
+ * @param[out] V optional right singular values of raft::device_matrix_view with layout
+ * raft::col_major
+ */
+template <typename ValueType, typename IndexType>
+void rsvd_fixed_rank(
+  const raft::handle_t& handle,
+  raft::device_matrix_view<ValueType, IndexType, raft::col_major> M,
+  raft::device_vector_view<ValueType, IndexType> S_vec,
+  IndexType p,
+  bool use_bbt,
```

Comment:
Would you consider putting algorithm configuration parameters (not input, output, or options like `p`, which are algorithm-independent) in a struct, rather than passing them in one by one?  That would reduce the number of arguments, make it easier for users to get default values, and would avoid common errors like mixing up the order of arguments.

Reply:
Looks like this function has been simplified quite a bit, which is great! An alternative to a struct might also be to break this apart into separate functions with names that might imply a specific configuration. What's not clear from the documentation at all is that `use_bbt` determines whether a rectangular or symmetric eigen decomposition is performed and `use_jacobi` determines whether Jacobi approx is used in place of QR (bbt=false) or divide-and-conquer (bbt=true). 

Maybe we could keep the `use_jacobi`, remove the `use_bbt` option, and set it to false for this particular one. Then can add a new function: `rsvd_fixed_rank_symmetric`. I think that makes the intention more clear and self explanatory. To make things even more clear, we could create `rsvd_fixed_rank_jacobi` and `rsvd_fixed_rank_symmetric_jacobi`. This would follow the convention used in `svd.cuh` as well for consistency.